src/lava/lib/optimization/utils/generators/mis.py

```diff
@@ -0,0 +1,180 @@
+# Copyright (C) 2021 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+# See: https://spdx.org/licenses/
+
+
+import numpy as np
+import numpy.typing as npt
+import networkx as netx
+
+from lava.lib.optimization.problems.problems import QUBO
+
+
+class MISProblem:
+    """
+    Utility class to instantiate Maximum Independent Set problems and
+    convert them to a QUBO formulation.
+    """
+
+    def __init__(self, num_vertices: int, connection_prob: float,
+                 seed: int = None):
+        """
+        Instantiate a new random MIS problem, with the given number of
+        vertices and connection probability.
+
+        :param num_vertices: Number of vertices of the random graph
+        :param connection_prob: Connection probability
+        :param seed: Seed for numpy random generator
+        """
+        self._num_vertices = num_vertices
+        self._connection_prob = connection_prob
+        self._seed = seed
+        self._adjacency = self._get_random_graph_mtx()
+
+    @property
+    def num_vertices(self) -> int:
+        """Returns the number of vertices in the graph."""
+        return self._num_vertices
+
+    @property
+    def connection_prob(self) -> float:
+        """Returns the connection probability of the graph."""
+        return self._connection_prob
+
+    @property
+    def seed(self):
+        """Returns the seed used for generating the graph."""
+        return self._seed
+
+    @staticmethod
+    def _get_graph_from_adjacency_matrix(adjacency_matrix):
+        num_vertices = adjacency_matrix.shape[0]
+        # create Graph
+        G = netx.Graph()
+        G.add_nodes_from(range(num_vertices))
+
+        for v1 in range(num_vertices):
+            for v2 in range(v1 + 1, num_vertices):
+                if adjacency_matrix[v1, v2] == 1:
+                    G.add_edge(v1, v2)
+        return G
+
+    @staticmethod
+    def _get_adjacency_of_complement_graph(
+            adjacency_matrix: np.ndarray) -> np.ndarray:
+        adj_mc = np.logical_not(adjacency_matrix)
+        adj_mc = adj_mc.astype(int)
+        adj_mc = adj_mc - np.diag(adj_mc.diagonal())
+        return adj_mc
+
+    @staticmethod
+    def _get_qubo_cost_from_adjacency(adjacency_matrix: np.ndarray,
+                                      w_diag: float,
+                                      w_off: float) -> np.ndarray:
+        if w_off <= 2 * w_diag:
+            raise ValueError(
+                "Off-diagonal weights must be > 2 x diagonal weights.")
+
+        # Translate the connectivity matrix to a QUBO matrix
+        num_variables = adjacency_matrix.shape[0]
+        q = - w_diag * np.eye(num_variables) + w_off / 2 * adjacency_matrix
+        return q.astype(int)
+
+    def get_graph(self) -> netx.Graph:
+        """Returns the graph in networkx format."""
+        graph = self._get_graph_from_adjacency_matrix(self._adjacency)

```

Comment:
Is there a reason to split get_graph and _get_graph_from_adjacency_matrix? Imho, you can delete a method if it's sole purpose is to call another method.

Reply:
The method `_get_graph_from_adjacency_matrix` is also used to get the networksx graph of the complement graph, the idea is to reduce code duplication. What do you think?

Reply:
Hm... It's at the edge of trivial. But ok, I see your point. And this method will be hidden to the user, anyway. Thus, feel free to leave it in.